core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -1721,7 +1733,7 @@ abstract class CoreReaderFragment :
     closeAllTabsButton?.isClickable = isClickable
   }
 
-  private fun restoreDeletedTabs() {
+  private suspend fun restoreDeletedTabs() {
```

Comment:
There is also a leg while restoring all tabs if closed(which is obvious since it is preparing the zimFileReader on the IO thread). But the user is not aware of it so we should show some progress/something so that the user can know about what is going on. Also, if we move to any other fragment while in this process then the application crashes.


https://github.com/user-attachments/assets/5149e3d9-449f-4332-8f62-7e22db981184



```kotlin
FATAL EXCEPTION: main
                                                                                                    Process: org.kiwix.kiwixmobile, PID: 18742
                                                                                                    java.util.NoSuchElementException: List is empty.
                                                                                                    	at kotlin.collections.CollectionsKt___CollectionsKt.last(_Collections.kt:418)
                                                                                                    	at org.kiwix.kiwixmobile.core.main.CoreReaderFragment.restoreDeletedTabs(CoreReaderFragment.kt:1746)
                                                                                                    	at org.kiwix.kiwixmobile.core.main.CoreReaderFragment.access$restoreDeletedTabs(CoreReaderFragment.kt:177)
                                                                                                    	at org.kiwix.kiwixmobile.core.main.CoreReaderFragment$restoreDeletedTabs$1.invokeSuspend(Unknown Source:18)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:808)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:101)
                                                                                                    	at android.os.Looper.loop(Looper.java:166)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7425)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
```